test/addons-abi/test_buffer/test_buffer.cc

```diff
@@ -0,0 +1,141 @@
+#include <string.h>
+#include <string>
+#include <node_api_helpers.h>
+
+#define JS_ASSERT(env, assertion, message) \
```

Comment:
I'm not certain about how generically useful it is, because it can only be called from a binding, which returns void.

I have a pile of macros for common operations [here](https://github.com/gabrielschulhof/iotivity-node/blob/f09b6ae0c51d8b653bd32f51b72ac5d3b4c3ec73/src/common.h). If you think any of them could be generically useful, I will add them.